### PR TITLE
lightning: refactor for reuse in load data part4

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### What is changed and how it works?
- separate dupe detect from backend
- make `split data into engines/chunks` and `local backend` independent on lightning config, rename some configs to better reflect its meaning and add comments.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/local/duplicate.go [hunk 1]
```diff
@@ -895,3 +907,174 @@ func (m *DuplicateManager) CollectDuplicateRowsFromTiKV(ctx context.Context, imp
...
+	// on TiKV, it is the max number of regions being checked concurrently
+	dupeConcurrency     int
+	duplicateDB         *pebble.DB
+	keyAdapter          KeyAdapter
+	importClientFactory ImportClientFactory
+}
+
+// CollectLocalDuplicateRows collect duplicate keys from local db. We will store the duplicate keys which
+// may be repeated with other keys in local data source.
+func (local *DupeController) CollectLocalDuplicateRows(ctx context.Context, tbl table.Table, tableName string, opts *encode.SessionOptions) (hasDupe bool, err error) {
```
Comment (D3Hunter) [inline br/pkg/lightning/backend/local/duplicate.go:927]: moved from `local` backend

**Comment by D3Hunter** [comment]
@lance6716 @GMHDBJD

File: br/pkg/lightning/backend/local/duplicate.go [hunk 2]
```diff
@@ -393,12 +401,12 @@ type DuplicateManager struct {
...
 	concurrency int
-	hasDupe     *atomic.Bool
+	hasDupe     atomic.Bool
 	indexID     int64
 }
 
-// NewDuplicateManager creates a new DuplicateManager.
-func NewDuplicateManager(
+// NewDupeDetector creates a new DupeDetector.
+func NewDupeDetector(
```
Comment (lance6716) [inline br/pkg/lightning/backend/local/duplicate.go:409]: also cc @gozssky

**Review by lance6716 — COMMENTED** [review]
13 / 28 files viewed

File: br/pkg/lightning/backend/local/duplicate.go (hunk 2 above)
Comment (gozssky) [reply br/pkg/lightning/backend/local/duplicate.go:409]: LGTM.

**Comment by D3Hunter** [comment]
ci pass 
https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_br_test/detail/tidb_ghpr_integration_br_test/12504/pipeline

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**